source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -115,12 +112,42 @@ void UdpStatsdSink::onHistogramComplete(const Stats::Histogram& histogram, uint6
   // are timers but record in units other than milliseconds, it may make sense to scale the value to
   // milliseconds here and potentially suffix the names accordingly (minus the pre-existing ones for
   // backwards compatibility).
-  const std::string message(absl::StrCat(prefix_, ".", getName(histogram), ":",
-                                         std::chrono::milliseconds(value).count(), "|ms",
-                                         buildTagStr(histogram.tags())));
+  const std::string message =
+      buildMessage(histogram, std::chrono::milliseconds(value).count(), "|ms");
   tls_->getTyped<Writer>().write(message);
 }
 
+const std::string UdpStatsdSink::buildMessage(const Stats::Metric& metric, uint64_t value,
+                                              const std::string& type) const {
+  switch (tag_format_.tag_position) {
+  case Statsd::TagPosition::TagAfterValue: {
+    const std::string message = absl::StrCat(
+        // metric name
+        prefix_, ".", getName(metric),
+        // value and type
+        ":", value, type,
+        // tags
+        buildTagStr(metric.tags()));
+    return message;
+  }
+
+  case Statsd::TagPosition::TagAfterName: {
+    const std::string message = absl::StrCat(
+        // metric name
+        prefix_, ".", getName(metric),
+        // tags
+        buildTagStr(metric.tags()),
+        // value and type
+        ":", value, type);
+    return message;
+  }
+
+  default:
+    // unreachable
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  }
```

Comment:
You shouldn't need the default as every branch is covered. If this is still failing to compile, move the `NOT_REACHED_GCOVR_EXCL_LINE` to the end of the function.

Reply:
Done. Do you think I should remove the `default` [here](https://github.com/envoyproxy/envoy/pull/16703/files#diff-a50f6997e1b3871b731aeb52e776bf91449bdc5996227ccbc89f3e62b91827b6R39) as well?